src/asInput/README.md

```diff
@@ -0,0 +1,35 @@
+# asInput
```

Comment:
@arizzitano I wouldn't mind discussing this component a bit more -- especially the `disabled` and `required` props :)

Reply:
@fysheets to address `disabled` and `required` -- although they aren't used directly by `asInput` itself (you probably saw the eslint ignore rule at the top of the file about this), components that extend `asInput` do use them (e.g. [InputText](https://github.com/edx/paragon/blob/master/src/InputText/index.jsx#L19-L20)) and they are passed along within the [`{...this.props}`](https://github.com/edx/paragon/blob/master/src/asInput/index.jsx#L95) spread on the `WrappedComponent`. My primary reason for adding them here was so that they'd be available within `inputProps` for any input component to use. Does that make sense?